spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/integration/filters/GcsLastModifiedFileListFilter.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.storage.integration.filters;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import com.google.cloud.storage.BlobInfo;
+
+import org.springframework.integration.file.filters.DiscardAwareFileListFilter;
+import org.springframework.integration.file.filters.FileListFilter;
+import org.springframework.lang.Nullable;
+
+/**
+ * The {@link FileListFilter} implementation to filter those files which
+ * {@link BlobInfo#getUpdateTime()} is less than the {@link #age} in comparison with the
+ * current time.
+ * <p>
+ * When {@link #discardCallback} is provided, it called for all the rejected files.
+ *
+ * @author Hosain Al Ahmad
+ */
+public class GcsLastModifiedFileListFilter implements DiscardAwareFileListFilter<BlobInfo> {
+
+	private static final long ONE_SECOND = 1000;
+
+	private final long age;
+
+	@Nullable
+	private Consumer<BlobInfo> discardCallback;
+
+	/**
+	 * Construct a {@link GcsLastModifiedFileListFilter} instance with provided {@link #age}.
+	 * Defaults to 60 seconds.
+	 *
+	 * @param age the age value.
+	 * @param unit the uunit for the age value.
+	 */
+	public GcsLastModifiedFileListFilter(long age, TimeUnit unit) {
```

Comment:
Can you use `java.time.Duration` instead?

Reply:
Done.